cg/meta/archive/archive.py

```diff
@@ -40,33 +35,28 @@ def filter_files_on_archive_location(
     ]
 
 
+ARCHIVE_HANDLERS: Dict[str, Type[ArchiveHandler]] = {
+    ArchiveLocations.KAROLINSKA_BUCKET: DDNDataFlowClient
+}
+
+
 class SpringArchiveAPI:
     """Class handling the archiving of sample SPRING files to an off-premise location for long
     term storage."""
 
     def __init__(
-        self,
-        ddn_dataflow_client: DDNDataFlowClient,
-        housekeeper_api: HousekeeperAPI,
-        status_db: Store,
+        self, housekeeper_api: HousekeeperAPI, status_db: Store, data_flow_config: DataFlowConfig
     ):
         self.housekeeper_api: HousekeeperAPI = housekeeper_api
-        self.ddn_client: DDNDataFlowClient = ddn_dataflow_client
         self.status_db: Store = status_db
-        self.handler_map: Dict[str, ArchiveModels] = {
-            ArchiveLocations.KAROLINSKA_BUCKET: ArchiveModels(
-                file_model=MiriaFile.from_file_and_sample, handler=self.ddn_client
-            )
-        }
+        self.data_flow_config: DataFlowConfig = data_flow_config
 
-    def call_corresponding_archiving_method(
-        self, files: List[FileTransferData], archive_location: ArchiveLocations
-    ) -> int:
-        return self.handler_map[archive_location].handler.archive_folders(
-            sources_and_destinations=files
-        )
+    def archive_files(self, files: List[FileAndSample], archive_location: ArchiveLocations) -> int:
+        archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[archive_location](self.data_flow_config)
```

Comment:
Will the archive location always be the same for this class? If so, maybe the archive handler could be provided in the constructor instead

Reply:
I'm guessing it won't always be the same though

Reply:
No, it won't be the same, although there is a one-to-one correspondence between archive_location and ArchiveHandlers at the moment. So the effect is the same as giving the ArchiveHandler, it is just a matter of where the instantiation happens.